connect/runtime/src/test/java/org/apache/kafka/connect/integration/InternalTopicsIntegrationTest.java

```diff
@@ -169,7 +164,6 @@ public void testFailToStartWhenInternalTopicsAreNotCompacted() throws Interrupte
         // Start the brokers but not Connect
         log.info("Starting {} Kafka brokers, but no Connect workers yet", numBrokers);
         connect.start();
```

Comment:
Oh interesting, this sets numWorkers=0, and therefore can call the blocking start() method safely.

WDYT about changing testFailToCreateInternalTopicsWithMoreReplicasThanBrokers to use the same pattern? That would eliminate the need for the non-blocking start method, and simplify the control flow.

The minority of call-sites are going to expect the workers to fail to start up, so I think it's okay to use a workaround instead of giving them a first-class method.

Reply:
I'm happy to rework `testFailToCreateInternalTopicsWithMoreReplicasThanBrokers`, but just in case someone downstream is using `start` in a way that relies on existing behavior, I'd also like to keep in the non-blocking method, even if we don't use it in any of our tests. I know we're not under any obligation to but it's such a small change that the negligible maintenance burden seems worth the tradeoff. Does that work for you?

Reply:
That's a great point. I've removed the overloaded variant and tweaked the only remaining integration test that uses it. That test itself is actually subtly broken since it doesn't (and never did) synchronously wait for worker startup before checking that internal topics don't exist, but since this PR doesn't introduce a regression I've thrown a TODO into the codebase and left everything else for a follow-up.